test/e2e/metrics_test.go

```diff
@@ -86,36 +103,63 @@ func NewMetricsTestConfig(t *testing.T, client, selector, clusterRole, clusterBi
 	}
 }
 
-// run will execute all steps of those tests
+// run will execute all steps of those tests with retries.
 func (c *MetricsTestConfig) run() {
-	c.createMetricsClusterRoleBinding()
-	token := c.getServiceAccountToken()
-	c.createCurlMetricsPod()
-	c.validate(token)
+	err := retry(func() error {
+		return c.createMetricsClusterRoleBinding()
+	})
+	require.NoError(c.t, err, "Failed to create ClusterRoleBinding after retries")
+
+	var token string
+	err = retry(func() error {
+		var err error
+		token, err = c.getServiceAccountToken()
+		if token == "" {
+			return fmt.Errorf("empty token")
+		}
+		return err
+	})
+	require.NoError(c.t, err, "Failed to get ServiceAccount token after retries")
+
+	err = retry(func() error {
+		return c.createCurlMetricsPod()
+	})
+	require.NoError(c.t, err, "Failed to create curl pod after retries")
+
+	err = retry(func() error {
+		return c.validate(token)
+	})
+	require.NoError(c.t, err, "Failed to validate metrics endpoint after retries")
+
```

Comment:
This feels very brute force and makes me think we don't actually understand the root cause of the problem yet.

I would not expect to need to retry really _any_ of this so long as we validate that the correct preconditions exist before proceeding to the next step.

Do we know where the flake failure is actually happening? Is that flake failure consistent?

Reply:
I'll add more detail to my mult-e2e-test run script to find out.

Reply:
now logging fails on my multi-run-e2e script:

https://pastebin.com/WMjR9nyV

Reply:
bad news is, it's still panic failing even with the heavy handed attempts to not

Reply:
I think the way that we set up things is that we start, and we run the tests,
then the metrics run in parallel. Sometimes, it seems that we clean up the cluster before the metrics tests finish.

Reply:
after the most recent commit, with the wait on cleanup:

```bash
Running test iteration 1/10...
Iteration 1: PASS
Running test iteration 2/10...
Iteration 2: PASS
Running test iteration 3/10...
Iteration 3: PASS
Running test iteration 4/10...
Iteration 4: PASS
Running test iteration 5/10...
Iteration 5: PASS
Running test iteration 6/10...
Iteration 6: PASS
Running test iteration 7/10...
Iteration 7: PASS
Running test iteration 8/10...
Iteration 8: PASS
Running test iteration 9/10...
Iteration 9: PASS
Running test iteration 10/10...
Iteration 10: PASS
Summary: Passed 10/10 times.
```
I am going to revise the PR to just do what the last commit does.

Reply:
I backed out these timing items in the most recent commit.